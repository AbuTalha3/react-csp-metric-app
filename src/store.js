/* eslint-disable import/no-extraneous-dependencies */
import { configureStore } from '@reduxjs/toolkit';
import { stockSlice } from './redux/homeSlice';

export default configureStore({
  reducer: {
    stocks: stockSlice.reducer,
  },
});
