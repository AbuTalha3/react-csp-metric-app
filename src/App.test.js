import { render, screen, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { BrowserRouter, MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import Home from './components/home';
import store from './store';
import App from './App';

describe('tests rect-csp-metric-app', () => {
  test('rect-csp-metric-render app component correctly', () => {
    render(
      <Provider store={store}>
        <BrowserRouter>
          <App />
        </BrowserRouter>
        ,
      </Provider>,
    );
  });

  test('render home page component correctly', () => {
    const component = renderer.create(
      <Provider store={store}>
        <MemoryRouter>
          <Home />
        </MemoryRouter>
      </Provider>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
