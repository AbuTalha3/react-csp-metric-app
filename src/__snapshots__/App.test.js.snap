// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests rect-csp-metric-app render details page component correctly 1`] = `<details />`;

exports[`tests rect-csp-metric-app render home page component correctly 1`] = `
<h1>
  Loading...
</h1>
`;

exports[`tests rect-csp-metric-app render home page component correctly 2`] = `
<h1>
  Loading...
</h1>
`;
